pykep/trajopt/gym/_solar_orbiter.py

```diff
@@ -0,0 +1,365 @@
+
+    def _decode_tofs(self, x):
+        if self._tof_encoding == "alpha":
+            # decision vector is  [t0, T, a1, a2, ....]
+            T = np.log(x[2:-2])
+            return T / sum(T) * x[1]
+        elif self._tof_encoding == "direct":
+            # decision vector is  [t0, T1, T2, T3, ... ]
+            return x[1:-2]
+        elif self._tof_encoding == "eta":
+            # decision vector is  [t0, n1, n2, n3, ... ]
+            dt = self.tof
+            T = [0] * self._n_legs
+            T[0] = dt * x[1]
+            for i in range(1, len(T)):
+                T[i] = (dt - sum(T[:i])) * x[i + 1]
+            return T
+
+    def _compute_dvs(self, x):
+        # 1 -  we 'decode' the times of flights and compute epochs (mjd2000)
+        T = self._decode_tofs(x)  # [T1, T2 ...]
+        ep = np.insert(T, 0, x[0])  # [t0, T1, T2 ...]
+        ep = np.cumsum(ep)  # [t0, t1, t2, ...]
+        # 2 - we compute the ephemerides
+        r = [0] * len(self._seq)
+        v = [0] * len(self._seq)
+        for i in range(len(self._seq)):
+            r[i], v[i] = self._seq[i].eph(ep[i])
+        # 3 - we solve the lambert problems
+        l = list()
+        for i in range(self._n_legs):
+            l.append(lambert_problem(
+                r[i], r[i + 1], T[i] * DAY2SEC, self._common_mu, False, 0))
+        # 4 - we compute the various dVs needed at fly-bys to match incoming
+        # and outcoming
+        DVfb = list()
+        for i in range(len(l) - 1):
+            vin = [a - b for a, b in zip(l[i].get_v2()[0], v[i + 1])]
+            vout = [a - b for a, b in zip(l[i + 1].get_v1()[0], v[i + 1])]
+            DVfb.append(fb_vel(vin, vout, self._seq[i + 1]))
+        return (DVfb, l, ep)
+
+    # Objective function
+    def fitness(self, x):
+        DVfb, lamberts, ep = self._compute_dvs(x)
+        if self._tof_encoding == "direct":
+            T = sum(x[1:-2])
+        elif self._tof_encoding == "alpha":
+            T = x[1]
+        elif self._tof_encoding == "eta":
+            T = sum(self.eta2direct(x)[1:-2])
+
+        # compute launch velocity and declination
+        Vinfx, Vinfy, Vinfz = [
+            a - b for a, b in zip(lamberts[0].get_v1()[0], self._seq[0].eph(ep[0])[1])
+        ]
+        Vinf_launch = sqrt(Vinfx ** 2 + Vinfy ** 2 + Vinfz ** 2)
+        # We transform it (only the needed component) to an equatorial system rotating along x
+        # (this is an approximation, assuming vernal equinox is roughly x and the ecliptic plane is roughly xy)
+        earth_axis_inclination = 0.409072975
+        Vinfz = - Vinfy * sin(earth_axis_inclination) + Vinfz * cos(earth_axis_inclination)
+        # And we find the vinf declination (in degrees)
+        sindelta = Vinfz / Vinf_launch
+        declination = asin(sindelta) / np.pi * 180.0
+        # We now have the initial mass of the spacecraft
+        m_initial = launchers.atlas501(Vinf_launch / 1000.0, declination)
```

Comment:
I think they use a different launcher  Atlas V 411 maybe ESOC can give you the table vinf decl mass for this and we can add a launcher?

Reply:
ESOC gave me a table with vinf and mass, but not using declinations. Added it to the launchers.